pkg/skaffold/test/test_factory.go

```diff
@@ -113,42 +113,33 @@ func (t FullTester) Test(ctx context.Context, out io.Writer, bRes []build.Artifa
 
 func (t FullTester) runTests(ctx context.Context, out io.Writer, bRes []build.Artifact) error {
 	for _, test := range t.testCases {
-		if err := t.runStructureTests(ctx, out, bRes, test); err != nil {
-			return fmt.Errorf("running structure tests: %w", err)
+		fqn, found := resolveArtifactImageTag(test.ImageName, bRes)
+		if !found {
+			logrus.Debugln("Skipping tests for", test.ImageName, "since it wasn't built")
+			return nil
 		}
-	}
-
-	return nil
-}
 
-func (t FullTester) runStructureTests(ctx context.Context, out io.Writer, bRes []build.Artifact, tc *latest.TestCase) error {
-	if len(tc.StructureTests) == 0 {
-		return nil
-	}
-
-	fqn, found := resolveArtifactImageTag(tc.ImageName, bRes)
-	if !found {
-		logrus.Debugln("Skipping tests for", tc.ImageName, "since it wasn't built")
-		return nil
-	}
+		if imageIsLocal, err := t.imagesAreLocal(test.ImageName); err != nil {
+			return err
+		} else if !imageIsLocal {
+			// The image is remote so we have to pull it locally.
+			// `container-structure-test` currently can't do it:
+			// https://github.com/GoogleContainerTools/container-structure-test/issues/253.
+			if err := t.localDaemon.Pull(ctx, out, fqn); err != nil {
+				return fmt.Errorf("unable to docker pull image %q: %w", fqn, err)
```

Comment:
Add an error code here. `TEST_IMG_PULL_ERR`

```
sErrors.NewError(err, 
   proto.ActionableErr{
     Message: fmt.Sprintf("unable to docker pull image %q: %w", fqn, err),
     ErrCode: proto.StatusCode_TEST_IMG_PULL_ERR,
 	  },
   }))
```

Reply:
Added the error code in pr #5385 and integrated the changes.